From 738bdaf792f9637bb05e5b8abdfa8637a7a47acf Mon Sep 17 00:00:00 2001
From: Laurent Pinchart <laurent.pinchart@ideasonboard.com>
Date: Wed, 22 May 2013 16:28:59 +0200
Subject: [PATCH 287/509] dma: xilinx: vdma: Used managed I/O mapping function

https://github.com/analogdevicesinc/linux.git xcomm_zynq
commit 0161185b030c9f78927133b85f2f25ac902e9239

Replace the of_iomap() call with a combination of
platform_get_resource() and devm_request_and_ioremap().

Signed-off-by: Laurent Pinchart <laurent.pinchart@ideasonboard.com>
Acked-by: Srikanth Thokala <sthokal@xilinx.com>
Signed-off-by: Zhong Hongbo <hongbo.zhong@windriver.com>
---
 drivers/dma/xilinx/xilinx_axivdma.c |   14 ++++++--------
 1 files changed, 6 insertions(+), 8 deletions(-)

diff --git a/drivers/dma/xilinx/xilinx_axivdma.c b/drivers/dma/xilinx/xilinx_axivdma.c
index 1b30a97..2a5621b 100644
--- a/drivers/dma/xilinx/xilinx_axivdma.c
+++ b/drivers/dma/xilinx/xilinx_axivdma.c
@@ -1220,6 +1220,7 @@ static int xilinx_vdma_of_probe(struct platform_device *op)
 {
 	struct xilinx_vdma_device *xdev;
 	struct device_node *child, *node;
+	struct resource *io;
 	int err, i;
 	const __be32 *value;
 	int num_frames = 0;
@@ -1238,12 +1239,11 @@ static int xilinx_vdma_of_probe(struct platform_device *op)
 	node = op->dev.of_node;
 	xdev->feature = 0;
 
-	/* iomap registers */
-	xdev->regs = of_iomap(node, 0);
-	if (!xdev->regs) {
-		dev_err(&op->dev, "unable to iomap registers\n");
-		return -ENOMEM;
-	}
+	/* Request and map I/O memory */
+	io = platform_get_resource(op, IORESOURCE_MEM, 0);
+	xdev->regs = devm_ioremap_resource(&op->dev, io);
+	if (IS_ERR(xdev->regs))
+		return PTR_ERR(xdev->regs);
 
 	/* Axi VDMA only do slave transfers */
 	if (of_device_is_compatible(node, "xlnx,axi-vdma")) {
@@ -1314,8 +1314,6 @@ static int xilinx_vdma_of_remove(struct platform_device *op)
 			xilinx_vdma_chan_remove(xdev->chan[i]);
 	}
 
-	iounmap(xdev->regs);
-
 	return 0;
 }
 
-- 
1.7.5.4

